tailscale/provider.go

```diff
@@ -31,6 +31,11 @@ func Provider(options ...ProviderOption) *schema.Provider {
 				DefaultFunc: schema.EnvDefaultFunc("TAILSCALE_TAILNET", nil),
 				Required:    true,
 			},
+			"base_url": {
+				Type:        schema.TypeString,
+				DefaultFunc: schema.EnvDefaultFunc("TAILSCALE_BASE_URL", "https://api.tailscale.com"),
+				Required:    true,
```

Comment:
```suggestion
```

I don't know exactly what the behaviour will be if required is set to true and there's a default, so I don't think this should be a required field.

Reply:
I have tried setting it to `false` in the past and removing it just now. Both yield the same result.

Either will prompt for the `base_url` if you do not specify in the provider block.

```
kyang@kyang terraform % terraform plan
provider.tailscale.base_url
  Enter a value: https://api.tailscale.com

╷
│ Error: InternalValidate
│
│   with provider["kyang.local/testing/tailscale"],
│   on <input-prompt> line 1:
│   (source code not available)
│
│ Internal validation of the provider failed! This is always a bug
│ with the provider itself, and not a user issue. Please report
│ this bug:
│
│ 1 error occurred:
│ 	* base_url: One of optional, required, or computed must be set
│
│
╵
```

Only setting it to require to true seems to work the way intended. If you do not specify the `base_url`, it will use the default.

Reply:
There should also be an `Optional` field you can set to true I think. I think you have to be explicit if a field is optional or required. A value of `false` in the `Required` field on its own does not indicate to the SDK that it is optional

Reply:
https://github.com/hashicorp/terraform-plugin-sdk/blob/v2.14.0/helper/schema/schema.go#L44

Here's that type in particular. The `Optional` field should be settable here

Reply:
Looks good. Tested locally.